concordium-std-derive/src/lib.rs

```diff
@@ -81,14 +103,14 @@ pub fn init(attr: TokenStream, item: TokenStream) -> TokenStream {
         quote! {
             #[export_name = #wasm_export_fn_name]
             pub extern "C" fn #rust_export_fn_name(amount: Amount) -> i32 {
-                use concordium_std::{Logger, trap};
+                use concordium_std::{trap, ExternContext, InitContextExtern, ContractState};
+                #setup_fn_optional_args
                 let ctx = ExternContext::<InitContextExtern>::open(());
-                let mut logger = Logger::init();
-                match #fn_name(&ctx, amount, &mut logger) {
+                match #fn_name(&ctx, #(#fn_optional_args),*) {
```

Comment:
To be helpful, we should check that the number of arguments the macro is applied to matches the number we will provide, and fail if this is not the case.

The error message could include the exact arguments we expect, which we do know at  this point.